### ISSUE TITLE
Fixed-point Softmax() calls exp_on_negative_values() twice

### ISSUE THREAD

**Issue by cousteaulecommandant** [tensorflow/tensorflow#88861]
Fixed-point `Softmax()` calls `exp_on_negative_values()` twice per element: once to calculate the value of the exp of each element
https://github.com/tensorflow/tensorflow/blob/4266e9ab53bb3f7fca0dd816b965b82862b69b4e/tensorflow/lite/kernels/internal/reference/softmax.h#L131
and another one a few lines earlier to calculate the sum of all exps
https://github.com/tensorflow/tensorflow/blob/4266e9ab53bb3f7fca0dd816b965b82862b69b4e/tensorflow/lite/kernels/internal/reference/softmax.h#L109-L110
The `exp_on_negative_values()` function can be rather slow, taking most of the time of `Softmax()` (which can be especially concerning in embedded implementations), so calling it twice per element makes the whole `Softmax()` function be about twice as slow as necessary.

Could this be optimized so that it is only called once per element?  Store the exp results in a temporary array (for example, reusing `output_data`), computing the sum as they're stored, and then applying the scaling to the precomputed exp results rather than computing them again.
(`SoftmaxInt16()` seems to follow a similar approach.)

**Comment by AdithyaSrivastava01** [comment]
Hey @Venkat6871 can you assign this issue to me?

**Comment by aravindmurali22** [comment]
Hi @cousteaulecommandant , thanks for the issue and PR.

@suleshahid , could you please review this? thanks.

**Comment by cousteaulecommandant** [comment]
Note that the PR was by @AdithyaSrivastava01 :)

**Comment by mihaimaruseac** [comment]
Sadly the PR was closed as stale. It couldn't be merged due to numerical issues: https://github.com/tensorflow/tensorflow/pull/89441#issuecomment-2748492711

**Comment by aravindmurali22** [comment]
Hi @cousteaulecommandant, as mentioned by @mihaimaruseac, since the PR couldn't be merged, I am closing this issue to keep things organized. Please feel free to try out [LiteRT](https://github.com/google-ai-edge/LiteRT) (earlier TFLite) and let us know if you run into any further issues. Thanks.
